src/arviz_base/reorg.py

```diff
@@ -488,34 +488,29 @@ def references_to_dataset(references, ds, sample_dims=None):
         return xr.full_like(aux_ds, references, dtype=np.array(references).dtype)
     # for array-like convert to dict so it is handled later on
     if isinstance(references, list | tuple | np.ndarray):
-        if np.ndim(references) > 1:
-            raise ValueError(
-                "Only 1D arrays are allowed. To generate more complex reference datasets "
-                "the xarray.Dataset constructor should be used."
-            )
         references = {var_name: references for var_name in ds.data_vars}
     if isinstance(references, dict):
         ref_dict = {}
         for var_name, da in ds.items():
             if var_name not in references:
                 continue
             ref_values = np.atleast_1d(references[var_name])
-            if np.ndim(ref_values) > 1:
-                raise ValueError(
-                    f"Only 1D arrays are allowed but the values for {var_name} variable have "
-                    "more dimensions. To generate more complex reference datasets the "
-                    "xarray.Dataset constructor should be used."
-                )
+            new_dims = ref_values.shape
+            new_dim_names = [f"ref_dim_{i}" for i in range(len(new_dims))]
```

Comment:
I would make `ref_dim` an argument of the function with `None` as default so it is `ref_dim` when data is 1d, `ref_dim_X` when n-d

Reply:
Yes, I think we give the users the option to input their `ref_dim`, they could pass in a list of `ref_dim` if they want to name it differently.
What do you think?